src/modules/thirdParty/sagas.ts

```diff
@@ -278,4 +284,25 @@ export function* thirdPartySaga(builder: BuilderAPI) {
       yield put(publishAndPushChangesThirdPartyItemsFailure(error.message)) // TODO: show to the user that something went wrong
     }
   }
+
+  function* handleConsumeSlotsRequest(action: ConsumeThirdPartyItemSlotsRequestAction) {
+    const { thirdPartyId, slots, merkleTreeRoot } = action.payload
+    try {
+      const maticChainId: ChainId = yield call(getChainIdByNetwork, Network.MATIC)
+      const thirdPartyContract: ContractData = yield call(getContract, ContractName.ThirdPartyRegistry, maticChainId)
+      const txHash: string = yield call(
+        sendTransaction as any,
+        thirdPartyContract,
+        'reviewThirdPartyWithRoot',
+        thirdPartyId,
+        merkleTreeRoot,
+        slots
+      )
+      // wait for the tx to be mined
+      waitForTx(txHash)
```

Comment:
shouldn't this be `yield`ed?
(awesome function btw)

oh also!  you could even remove the comment if you want, now that you named the function is super clear what's happening

Reply:
Indeed, it needs to be yielded, updated! thanks :)